src/pages/Scheduling/components/CreateScheduleTemplateSheet.tsx

```diff
@@ -235,6 +236,34 @@ export default function CreateScheduleTemplateSheet({
     );
   };
 
+  const validFrom = form.watch("valid_from");
+  const validTill = form.watch("valid_to");
+  const [dateValidationCallout, setDateValidationCallout] =
```

Comment:
why maintain a state for this? You shouldn't be storing a ReactNode inside state anyways.

Let's not complicate this - just have a state for rending the callout (and rename to something else since dateValidationCallout is generic - we are only rendering for a specific case anyways).